django/contrib/sitemaps/__init__.py

```diff
@@ -202,23 +208,25 @@ def _urls(self, page, protocol, domain):
 
             if self.i18n and self.alternates:
                 for lang_code in self._languages():
-                    loc = f"{protocol}://{domain}{self._location(item, lang_code)}"
-                    url_info["alternates"].append(
-                        {
-                            "location": loc,
-                            "lang_code": lang_code,
-                        }
-                    )
+                    if lang_code in self.get_languages_for_item(item[0]):
```

Comment:
Why don't do this in a single line? I think we can also call `get_languages_for_item()` only once:
```python
if self.i18n and self.alternates:
    languages = self.get_languages_for_item(item[0])
    for lang_code in languages:
        loc = f"{protocol}://{domain}{self._location(item, lang_code)}"
        ...

    if self.x_default and settings.LANGUAGE_CODE in languages:
        lang_code = settings.LANGUAGE_CODE
        loc = f"{protocol}://{domain}{self._location(item, lang_code)}"
        ...
```

Reply:
I followed your advice. If you have time, could you explain how 
```python
if self.x_default and settings.LANGUAGE_CODE in languages:
    lang_code = settings.LANGUAGE_CODE
```
is better than
```python
lang_code = settings.LANGUAGE_CODE
if self.x_default and lang_code in languages:
```

please ?

Reply:
One assignment less, not much difference, TBH.